crates/biome_js_analyze/src/lint/nursery/use_numeric_separators.rs

```diff
@@ -0,0 +1,279 @@
+                Some(State::InconsistentGrouping)
+            } else {
+                // Contains separators which are not present in the expected value.
+                Some(State::UnnecessaryGrouping)
+            }
+        } else {
+            // Missing separators entirely.
+            Some(State::UnreadableLiteral)
+        }
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+
+        match state {
+            State::UnreadableLiteral => Some(RuleDiagnostic::new(
+                rule_category!(),
+                node.range(),
+                markup!("Long numeric literal lacks separators"),
+            )),
+            State::InconsistentGrouping => Some(RuleDiagnostic::new(
+                rule_category!(),
+                node.range(),
+                markup!("Inconsistent grouping of digits in numeric literal"),
+            )),
+            State::UnnecessaryGrouping => Some(RuleDiagnostic::new(
+                rule_category!(),
+                node.range(),
+                markup!("Unnecessary grouping of digits in numeric literal"),
+            )),
+        }
+    }
+
+    fn action(ctx: &RuleContext<Self>, state: &Self::State) -> Option<JsRuleAction> {
+        let token = ctx.query().value_token().ok()?;
+        let num = format_num(token.text_trimmed());
+
+        let new_token = JsSyntaxToken::new_detached(token.kind(), &num, [], []);
+        let mut mutation = ctx.root().begin();
+        mutation.replace_token_transfer_trivia(token.clone(), new_token);
+        Some(JsRuleAction::new(
+            ctx.metadata().action_category(ctx.category(), ctx.group()),
+            ctx.metadata().applicability(),
+            match state {
+                State::UnreadableLiteral => markup! { "Add numeric separators" },
+                State::InconsistentGrouping => markup! { "Fix numeric separator grouping" },
+                State::UnnecessaryGrouping => markup! { "Remove unnecessary numeric separators" },
+            },
+            mutation,
+        ))
+    }
+}
+
+pub enum State {
+    UnreadableLiteral,
+    InconsistentGrouping,
+    UnnecessaryGrouping,
+}
+
+/// Add chunk separators to a number string, starting from the right.
+/// The "uneven" chunk is added to the left of the first separator.
+/// 1234567890 -> 1_234_567_890
+fn add_separators_from_right(num: &str, min_digits: usize, group_length: usize) -> String {
+    if num.len() < min_digits {
+        num.to_owned()
+    } else {
+        let mut result = String::new();
+        let mut count = 0;
+
+        for c in num.chars().rev() {
+            if count > 0 && count % group_length == 0 {
+                result.push('_');
+            }
+            result.push(c);
+            count += 1;
+        }
+
+        result.chars().rev().collect()
+    }
+}
+
+/// Add chunk separators to a number string, starting from the left. Used for fractional parts.
```

Comment:
It's unclear how `min_digits` and `group_length` are used. The documentation doesn't mention them